src/idiomatic/leveraging-the-type-system/borrow-checker-invariants/generalizing-ownership.md

```diff
@@ -0,0 +1,88 @@
+---
+minutes: 15
+---
+
+# Generalizing Ownership
+
+The logic of the borrow checker, while modelled off "memory ownership", can be
+abstracted away from that use case to model other problems where we want to
+prevent API misuse.
+
+```rust,editable,compile_fail
+// An internal data type to have something to hold onto.
+pub struct Internal;
+// The "outer" data.
+pub struct Data(Internal);
+
+fn shared_use(value: &Data) -> &Internal {
+    &value.0
+}
+fn exclusive_use(value: &mut Data) -> &mut Internal {
+    &mut value.0
+}
+fn deny_future_use(value: Data) {}
+
+fn main() {
+    let mut value = Data(Internal);
+    let deny_mut = shared_use(&value);
+    let try_to_deny_immutable = exclusive_use(&mut value); // ❌🔨
```

Comment:
If this line does not compile because of the function call, what does the variable declaration add? Why not let it simply be `exclusive_use(&mut value);`?

Similarly for the line below that also does not compile.